File: C++/medium/nth-prime/nth_prime.cpp
#include "nth_prime.h"

namespace nth_prime {
  bool prime(const int& num) {
    bool result{false};
    if (num > 1) {
      result = true;
      for (int i{2}; i <= (num / 2); ++i) {
        if (num % i == 0) {
          result = false;
          break;
        }
      }
    }
    return result;
  }

  int nth(const int& n) {
    if (n < 1) throw std::domain_error("there's no n primer number lower than 1");
    int result{2};
    for (int i{1}, j{2}; i <= n; ++i, ++j) {
      while (!prime(j)) ++j;
      result = j;
    }
    return result;
  }
}  // namespace nth_prime

File: Python/easy/nth-prime/nth_prime.py
def is_prime(n):
    if n <= 1:
        return False
    elif n == 2:
        return True
    elif n % 2 == 0:
        return False
    else:
        for i in range(3, int(n ** 0.5) + 1, 2):
            if n % i == 0:
                return False
        return True


def prime(number):
    if number == 0:
        raise ValueError("there is no zeroth prime")

    result = 1
    while number > 0:
        result += 1
        if is_prime(result):
            number -= 1
    
    return result
